types/resolve/index.d.ts

```diff
@@ -87,7 +87,16 @@ declare function resolveSync(id: string, opts?: resolve.SyncOpts): string;
  */
 declare function resolveIsCore(id: string): boolean | undefined;
 
+// https://github.com/Microsoft/TypeScript/issues/3496#issuecomment-128553540
+type JSONValue = string | number | boolean | JSONObject | JSONArray;
+interface JSONObject {
+    [x: string]: JSONValue;
+}
+interface JSONArray extends Array<JSONValue> { }
+
 declare namespace resolve {
+  export type PackageJSON = JSONObject;
```

Comment:
@ljharb Should it be exported as well? Perhaps? If so, is it correct to export it like this?

For instance, Jest could possibly use exported type here: https://github.com/facebook/jest/blob/a293b75310cfc209713df1d34d243eb258995316/packages/jest-resolve/src/defaultResolver.ts#L47-L50

Reply:
Seems fine to me, since `ReturnType<JSON.parse>` is useless